### ISSUE TITLE
Don't get import/order to work correctly with import aliases

### ISSUE THREAD

**Issue by jens-duttke** [import-js/eslint-plugin-import#1682]
No matter what  I try, I don't get the `import/order` plugin to work as expected.

These are the imports, and they are already ordered as expected:
```
import classNames from 'classnames';
import * as React from 'react';

import { I18N } from '@context/i18n';
import { SVGIcon } from '@ui';
import * as Utils from '@utils';

import * as styles from './styles.scss';
```

Anyhow, the rule complains about:
> `classnames` import should occur after import of `@utils`
> `react` import should occur after import of `@utils`

`classnames` and `react` are located in `node_modules`
Everything starting with `@` indicates an import alias (so I assume they should be interpreted as `internal`).

My eslint configuration looks like this:
```
{
	settings: {
		'import/parsers': {
			'@typescript-eslint/parser': ['.ts', '.tsx']
		},
		'import/internal-regex': '^@'
	},
	rules:  {
		'import/order': ['error', {
			'alphabetize': { order: 'asc', caseInsensitive: true },
			'groups': ['builtin', 'unknown', 'external', 'internal', 'parent', 'sibling', 'index'],
			'newlines-between': 'always'
		}]
	}
}
```
I've played around since multiple hours now, but don't get the expected result. The only thing which happens is, that it sometimes say that `@utils` should occur before `classnames`, and sometimes it says `classnames` should occur after `@utils` ...

What have I missed?

**Comment by xfournet** [comment]
@jens-duttke are you using version 2.20.1 ? i had similar problem that are not present up to 2.20.0 included

**Comment by ljharb** [comment]
Although using `@` is an unwise idea, given that npm packages can begin with a `@`-prefixed scope, your settings seem like they should work.

Any chance you'd want to switch to a sigil that doesn't collide with standard npm package naming conventions?

**Comment by jens-duttke** [comment]
I don't import/require npm packages starting with an `@`, so there is no collision.
Beside that, I've just tested to replace `@` by `~` (which is not a valid character in npm package names), but the exact same problem still occurs:

> `classnames` import should occur after import of `~utils` [Error/import/order]
> `react` import should occur after import of `~utils` [Error/import/order]

**Comment by ryprice** [comment]
I believe I'm having the same issue preventing me from using this rule. For all my ts projects, I import local modules using the fully-qualified name using a module alias. ([tsconfig](https://github.com/ryprice/ququmber-ui/blob/master/tsconfig.json#L14), [example import](https://github.com/ryprice/ququmber-ui/blob/master/src/button/UIButton.tsx#L3)). Even when I'm very explicit in `.eslintrc` about what group `ququmber-ui` should be, I see similar errors to @jens-duttke. 

```json
{
  "newlines-between": "ignore",
  "groups": [
    "builtin",
    "external",
    "internal",
    "parent",
    "sibling",
    "index"
    "unknown"
  ],
  "alphabetize": {"order": "asc", "caseInsensitive": true},
  "pathGroups": [
    {"pattern": "ququmber-ui/**", "group": "parent"},
    {"pattern": "listlab-api/**", "group": "internal"},
    {"pattern": "lodash", "group": "external"},
    {"pattern": "react", "group": "external"},
    {"__comment": ".....other external packages removed for brevity"}
  ]
}
```

Here is an example of an error I see:
```
D:\src\ququmber-ui\src\button\UIIconButton.tsx
 1:1  error  `react` import should occur after import of `ququmber-ui/utils/useDelayedMouseHover`         import/order
 2:1  error  `react-tether` import should occur after import of `ququmber-ui/utils/useDelayedMouseHover`  import/order
```

**Comment by ryprice** [comment]
I did a bit of debugging and figured out that, at least in my case, imports for `ququmber-ui` (the local package) are [being skipped](https://github.com/benmosher/eslint-plugin-import/blob/master/src/rules/order.js#L318) because `ququmber-ui` is considered external and `pathGroupsExcludedImportTypes` defaults to `["builtin", "external"]`.

My fix `"pathGroupsExcludedImportTypes": []` Hope this helps you @jens-duttke.

**Comment by jens-duttke** [comment]
@ryprice This works fine. Thanks!

**Comment by ludwig-pro** [comment]
I think there is a bug when ESLint load the config, so config-validator.js : 

```sh
Configuration for rule "import/order" is invalid:
	Value {"newlines-between":"always","groups":[["builtin","external"],["internal","index","sibling","parent"]],"alphabetize":{"order":"asc","caseInsensitive":true},"pathGroups":[{"pattern":"~/**","group":"internal"}],"pathGroupsExcludedImportTypes":[]} should NOT have additional properties.
```

**Comment by ljharb** [comment]
@ludwig-pro no bug, but you're probably looking at the docs on master instead of on the latest tagged release.

**Comment by ludwig-pro** [comment]
you are right, but I just notice that when I add `pathGroups` or `pathGroupsExcludedImportTypes` ESLint console throw me an error

```sh
(node:6627) UnhandledPromiseRejectionWarning: Error: .eslintrc.js:
	Configuration for rule "import/order" is invalid:
	Value {"groups":[["builtin","external"],["internal","index","sibling","parent"]],"pathGroupsExcludedImportTypes":[],"pathGroups":[{"pattern":"~/**","group":"internal"}],"newlines-between":"always"} should NOT have additional properties.
```

it's strange because when I inspect node_modules, I use last version who support this properties. 

```    
"eslint": "^7.19.0",
"eslint-config-prettier": "^6.11.0",
"eslint-plugin-import": "^2.22.1",
```

**Comment by ludwig-pro** [comment]
and the rule 

```
'import/order': [
      'error',
      {
        groups: [
          ['builtin', 'external'],
          ['internal', 'index', 'sibling', 'parent'],
        ],
        pathGroupsExcludedImportTypes: [],
        pathGroups: [{ pattern: '~/**', group: 'internal' }],
        'newlines-between': 'always',
      },
    ],
    ```

**Comment by ludwig-pro** [comment]
OK just restart vs code... and it's works 👍

**Comment by Slapbox** [comment]
@ljharb I think that the docs regarding when someone might want to use `pathGroupsExcludedImportTypes` could use clarification - though I'm not sure of exactly what it should say to be clear.

I spent quite a while trying to figure out why adding an `internal-regex` string was causing all of our monorepo packages to give the error `Relative imports from parent directories are not allowed` and was about to open an issue regarding that when I found [the golden comment from @ryprice](https://github.com/benmosher/eslint-plugin-import/issues/1682#issuecomment-608969468). 

Perhaps that particular result we saw is a bug? Because I don't understand why it marked package imports (`import something from 'our-package'`) as being relative imports from a parent directory. We saw that message whether we did or didn't define a resolver configuration.
